_topics/en/time-warp.md

```diff
@@ -0,0 +1,181 @@
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **Time warp** is an attack that allows a majority of miners to produce
+  up to one block per second indefinitely.  The attack could be used to
+  steal the remaining block subsidy or to perform a DoS attack on nodes
+  and clients.
+
+---
+
+Bitcoin blocks contain an nTime field in their headers.  We want this to
+be the time the block was created by its miner, but there's no way to
+directly enforce that.  At best, we can try to enforce that the nTime
+field is set to a value after any events it commits to and before the
+present time.  Two rules in the Bitcoin Protocol try to accomplish that:
+
+1. The nTime of a block must be greater than the median nTime of the
+   previous 11 blocks, called Median Time Past (MTP).
+
+2. The nTime of a block in UTC must be less than two hours in the future
+   according to the UTC clock on the computer of the local full node
+   verifying the block.
+
+The first rule works well when a majority of hash rate is reporting
+accurate times---it prevents a malicious or broken miner from reporting a
+time more than a few hours in the past under normal circumstances.
+However, a majority of hash rate can refuse to increment time by more
+than one second per block, producing new blocks whose times are
+increasingly further and further into the past.
+
+That shouldn't be a long-term problem.  If the first and last blocks in
+a 2,016-block *retarget period* are closer together than 20,160 minutes
+(i.e. an average of roughly ten minutes per block), <!-- roughly because
+of off-by-one error --> the Bitcoin Protocol raises the Proof of Work
+(PoW) difficulty and existing miners become less profitable, strongly
+disincentivizing such dishonest time reporting.  Unfortunately, because
+the protocol only looks at the difference between the first and last
+blocks, rather than taking a true average of the difference between each
+block and its parent, it's possible to manipulate the Difficulty
+Adjustment Algorithm (DAA).  For example, consider the following times
+for 2,016 blocks:
+
+|                | Block 0 | Block 1 |  ... | Block 2014 | Block 2015 |
+| Real time      | 0       | 600     |  ... | 1,208,400  | 1,209,000  |
+| Reported nTime | 0       | 1       |  ... | 2014       | 1,209,000  |
+
+Although the involved miners reported the minimum allowed change for
+every block except the last block, the algorithm makes the same
+difficulty change it would've had they reported accurate times (in this
+case, there's no difficulty change).  If the miner continues this
+behavior for another retarget period, they can actually lower
+difficulty:
+
+|                | Block 2016 | Block 2017 | ... | Block 4030 | Block 4031 |
+| Real time      | 1209600    | 1210200    | ... | 2418000    | 2418600    |
+| Reported nTime | 2016       | 2017       | ... | 4030       | 2418600    |
```

Comment:
The 2016 nTime here, it wouldn't be valid under rule 1, correct? Since block 2015's nTime was so large.